tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/oidc/OidcEndpointsInfoResolver.java

```diff
@@ -0,0 +1,103 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.james.jmap.oidc;
+
+import java.net.URL;
+import java.time.Instant;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
+import org.apache.james.core.Username;
+import org.apache.james.jmap.exceptions.UnauthorizedException;
+import org.apache.james.jwt.DefaultCheckTokenClient;
+import org.apache.james.jwt.introspection.IntrospectionEndpoint;
+import org.apache.james.jwt.introspection.TokenIntrospectionResponse;
+import org.apache.james.jwt.userinfo.UserinfoResponse;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.util.streams.Iterators;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+import com.linagora.tmail.james.jmap.model.Aud;
+import com.linagora.tmail.james.jmap.model.Sid;
+import com.linagora.tmail.james.jmap.model.Token;
+import com.linagora.tmail.james.jmap.model.TokenInfo;
+
+import reactor.core.publisher.Mono;
+
+public class OidcEndpointsInfoResolver implements TokenInfoResolver {
+    private static final String SID_PROPERTY = "sid";
+
+    private final DefaultCheckTokenClient checkTokenClient;
+    private final MetricFactory metricFactory;
+    private final URL userInfoURL;
+    private final IntrospectionEndpoint introspectionEndpoint;
+    private final JMAPOidcConfiguration configuration;
+
+    @Inject
+    public OidcEndpointsInfoResolver(DefaultCheckTokenClient checkTokenClient,
+                                     MetricFactory metricFactory,
+                                     @Named("userInfo") URL userInfoURL,
```

Comment:
How about introducing a strong type for `userInfoURL`?

Reply:
Not sure it's really necessary, as we reuse the CheckTokenClient api?

Reply:
This is ok to me too.